README.md

```diff
@@ -1,2 +1,59 @@
-# puppetlabs-ruby_task_helper
-A ruby library for running tasks
+# Ruby Task Helper
+
+This is a helper library for writing tasks. It provides a class that handles error generation, simplifies JSON input and output, and makes testing your task easier.
+
+#### Table of Contents
+
+1. [Description](#description)
+2. [Setup](#setup)
+3. [Usage](#usage)
+
+## Description
+
+## Setup
+
+Include this module in a [Puppetfile]():
+
+```
+mod puppetlabs-ruby_task_helper
+```
+
+## Usage
+
+When writing your task include the library in your script, extend the `TaskClass` module, and write the `task()` function. The `task()` function **must** take a hash of parameters as input (even if it's an empty hash), and **must** return a hash. The following is an example of a task that uses the library
+
+`mymodule/tasks/task.rb`
+```
+require_relative "#{ENV['_installdir']}/ruby_task_helper.rb"
+
+include TaskClass
+extend TaskClass
+
+def task(params = {})
+    puts "Hi! I'm #{params['name']}"
+end
+
```

Comment:
This will need to be gaurded for unit tests to work.

Reply:
I'm not sure what you mean by 'guarded'?

Reply:
`MyClass.run` shouldn't be invoked whenever you load the class.

In python_task_helper I added an [example test](https://github.com/puppetlabs/puppetlabs-python_task_helper/blob/master/tests/test_mytask.py) that loads the task as a library, instantiates the class, and runs it. If you tried to do that with the task here, it will create `MyClass` and call `run` when you load the task as a library.

The pattern in Ruby appears to be
```
if __FILE__ == $0
  # do thing
```